app/uk/gov/hmrc/eusubsidycompliancefrontend/controllers/UndertakingController.scala

```diff
@@ -175,6 +177,39 @@ class UndertakingController @Inject() (
     )
   }
 
+  def getAddEmailForVerification(status: String = EmailStatus.New.toString): Action[AnyContent] = enrolled.async {
+    implicit request =>
+      val backLink = EmailStatus.withName(status) match {
+        case EmailStatus.Unverified => routes.UnverifiedEmailController.unverifiedEmail.url
+        case _ => routes.UndertakingController.getSector.url
+      }
+      Future.successful(Ok(inputEmailPage(emailForm, backLink, Some(status))))
+  }
+
+  def postAddEmailForVerification(status: String = EmailStatus.New.toString): Action[AnyContent] = enrolled.async {
+    implicit request =>
+      val backLink = EmailStatus.withName(status) match {
```

Comment:
Except for .url things are the same. Could stick to a rule of three :)

```
is a code refactoring rule of thumb to decide when a replicated piece of code should be replaced by a new procedure. It states that you are allowed to copy and paste the code once, but that when the same code is replicated three times, it should be extracted into a new procedure.
```

![image](https://github.com/hmrc/eu-subsidy-compliance-frontend/assets/3098723/f836d988-ef64-4994-86b7-f2e50b1d3d6d)